### ISSUE TITLE
False positive on reportUninitializedInstanceVariable with dataclasses

### ISSUE THREAD

**Issue by ItsDrike** [DetachHead/basedpyright#491]
BasedPyright incorrectly marks variables declared in an abstract dataclass as uninitialized:

```python
from abc import ABC
from dataclasses import dataclass
from typing import final


@dataclass
class Foo(ABC):
    x: int
    y: int


@final
class Bar(Foo): ...  # basedpyright: Variables defined in abstract base class are not initialized in final class "Bar"
```

Interestingly enough, if the `Foo` dataclass doesn't inherit from `ABC` class, things work as expected and no error is reported.

**Comment by PerchunPak** [comment]
Dataclass children must use `@dataclass` too, then the issue disappears. I remember there are more bugs if you forget to add `@dataclass` to children, but cannot remember which exactly

**Comment by DetachHead** [comment]
probably related: #667

**Comment by DetachHead** [comment]
actually they are pretty different issues. #667 is specifically to do with abstract attributes on a metacaclass decorated with `@dataclass_transform`
